src/v/datalake/datalake_manager.cc

```diff
@@ -42,8 +44,124 @@ datalake_manager::datalake_manager(
   , _features(features)
   , _coordinator_frontend(frontend)
   , _as(as)
-  , _sg(sg) {}
+  , _sg(sg)
+  , _effective_max_translator_buffered_data(
+      std::min(memory_limit, max_translator_buffered_data))
+  , _parallel_translations(
+      size_t(
+        std::floor(memory_limit / _effective_max_translator_buffered_data)),
+      "datalake_parallel_translations")
+  , _translation_ms_conf(config::shard_local_cfg()
+                           .iceberg_translation_interval_ms_default.bind()) {
+    // Handle leadership changes
+    auto leadership_registration
+      = _group_mgr->local().register_leadership_notification(
+        [this](
+          raft::group_id group,
+          ::model::term_id,
+          std::optional<::model::node_id>) { on_group_notifiction(group); });
 
-ss::future<> datalake_manager::stop() { co_await _gate.close(); }
+    // Handle topic properties changes (iceberg.enabled=true/false)
+    auto topic_properties_registration
+      = _topic_table->local().register_ntp_delta_notification(
+        [this](cluster::topic_table::ntp_delta_range_t range) {
+            for (auto& entry : range) {
+                if (
+                  entry.type
+                  == cluster::topic_table_ntp_delta_type::properties_updated) {
+                    on_group_notifiction(entry.group);
+                }
+            }
+        });
+    _deregistrations.reserve(2);
+    _deregistrations.emplace_back([this, leadership_registration] {
+        _group_mgr->local().unregister_leadership_notification(
+          leadership_registration);
+    });
+    _deregistrations.emplace_back([this, topic_properties_registration] {
+        _topic_table->local().unregister_ntp_delta_notification(
+          topic_properties_registration);
+    });
+    _translation_ms_conf.watch([this] {
+        ssx::spawn_with_gate(_gate, [this]() {
+            for (const auto& [group, _] : _translators) {
+                on_group_notifiction(group);
+            }
+        });
+    });
+}
+
+ss::future<> datalake_manager::stop() {
+    auto f = _gate.close();
+    _deregistrations.clear();
+    co_await ss::max_concurrent_for_each(
+      _translators, 32, [](auto& entry) mutable {
+          return entry.second->stop();
+      });
+    co_await std::move(f);
+}
+
+void datalake_manager::on_group_notifiction(raft::group_id group) {
+    auto partition = _partition_mgr->local().partition_for(group);
+    if (!partition || !model::is_user_topic(partition->ntp())) {
+        return;
+    }
+    const auto& topic_cfg = _topic_table->local().get_topic_cfg(
+      model::topic_namespace_view{partition->ntp()});
+    if (!topic_cfg) {
+        return;
+    }
+    auto it = _translators.find(group);
+    // todo handle topic / partition disabling
```

Comment:
nit: I generally prefer doing something like `TODO(iceberg)` so it's clearly grouped to the project and makes it easy to grep for all the iceberg todos before we GA 😄